packages/lab/src/tabs-next/TabNext.tsx

```diff
@@ -0,0 +1,93 @@
+import { makePrefixer } from "@salt-ds/core";
+import {
+  forwardRef,
+  ReactElement,
+  ComponentPropsWithoutRef,
+  MouseEvent,
+  FocusEvent,
+  useEffect,
+} from "react";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+
+import tabCss from "./TabNext.css";
+import clsx from "clsx";
+import { OverflowItem } from "@fluentui/react-overflow";
+import { useTabs } from "./TabNextContext";
+
+const withBaseName = makePrefixer("saltTabNext");
+
+export interface TabNextProps extends ComponentPropsWithoutRef<"button"> {
+  /* Value prop is mandatory and must be unique in order for overflow to work. */
+  value: string;
+}
+
+export const TabNext = forwardRef<HTMLButtonElement, TabNextProps>(function Tab(
+  props,
+  ref
+): ReactElement<TabNextProps> {
+  const {
+    children,
+    className,
+    disabled: disabledProp,
+    onClick,
+    onFocus,
+    value,
+    ...rest
+  } = props;
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-tab-next",
+    css: tabCss,
+    window: targetWindow,
+  });
+  const {
+    isSelected,
+    select,
+    isFocused,
+    focus,
+    disabled: tabstripDisabled,
+    unregisterTab,
+    registerTab,
+  } = useTabs();
+  const selected = isSelected(value);
+  const focusable = isFocused(value);
+  const disabled = tabstripDisabled || disabledProp;
+
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    select(event);
+    onClick?.(event);
+  };
+
+  const handleFocus = (event: FocusEvent<HTMLButtonElement>) => {
+    focus(value);
+    onFocus?.(event);
+  };
+
+  useEffect(() => {
+    registerTab({ value, label: children });
```

Comment:
If the user passes jsx as children this will cause infinite loop, no? Why not store the references in a ref? Besides, why is this register/unregister even necessary? Can't we map/filter the children?

Reply:
Mapping/filtering the children is brittle. Have you tested the infinite loop theory? I tested the implementation and I didn't see any issues